### PR TITLE
Upgrade celery and django-celery

### PR DESCRIPTION
**Pull request by gcapalbo**

### CODE REVIEW

File: requirements/requirements.txt [hunk 1]
```diff
@@ -55,7 +55,7 @@ six==1.4.1
 socketpool==0.5.2
 markdown==2.2.1
 django-markup-deprecated
-amqp==1.0.13
+amqp==1.4.6
 amqplib==1.0.2
 sqlagg==0.2.2
 django-redis==3.3
```
Comment (gcapalbo) [inline requirements/requirements.txt]: Just pinning the versions here, all of these came by just doing:
    pip install --upgrade celery
    pip install --upgrade django-celery

**Comment by czue** [comment]
do we want to QA this?

**Comment by gcapalbo** [comment]
@snopoke @czue 

I didn't see anything we need to change based on http://celery.readthedocs.org/en/latest/whatsnew-3.1.html or https://launchpad.net/pytz/+announcements. I also locally tested flower, celery beat, and the major tasks invoked by the reminders and sms frameworks and all that worked fine after upgrading.

As Simon mentioned, django is supported out of the box now, but in order to take advantage of that we'd have to update our code to use the new api.  Since doing that isn't urgent it didn't seem worth doing now.

If we want to be safe, we could put this on staging for a bit and just see if anything falls out of it.  Otherwise, I think this should be good.

**Comment by czue** [comment]
cool, thanks. i'm happy to trust your judgment, but will let @snopoke chime in as well. it does seem like throwing it on staging for a bit doesn't hurt, though we won't exercise a lot of the code with celerybeat off anyways...

**Comment by gcapalbo** [comment]
yeah, I'm fine with putting it on staging for a week or two, just wasn't sure how quickly this was needed

**Comment by snopoke** [comment]
I'll deploy it to staging and make sure exports etc still work.

**Comment by snopoke** [comment]
Was getting this error on staging. For some reason it was using the celery database backend instead of the djcelery one.

I wasn't able to reproduce it locally but changing the localsettings on staging seems to have fixed it:
```python
#CELERY_RESULT_BACKEND = 'database'
CELERY_RESULT_BACKEND = 'djcelery.backends.database:DatabaseBackend'
```

Error:
```
  File "/home/cchq/www/staging/python_env/local/lib/python2.7/site-packages/django/core/handlers/base.py", line 115, in get_response
    response = callback(request, *callback_args, **callback_kwargs)

  File "/home/cchq/www/staging/python_env/local/lib/python2.7/site-packages/newrelic-2.40.0.34/newrelic/hooks/framework_django.py", line 497, in wrapper
    return wrapped(*args, **kwargs)

  File "/home/cchq/www/staging/python_env/local/lib/python2.7/site-packages/django/contrib/auth/decorators.py", line 25, in _wrapped_view
    return view_func(request, *args, **kwargs)

  File "submodules/django-soil-src/soil/views.py", line 41, in ajax_job_poll
    context = get_download_context(download_id, check_state=True)

  File "submodules/django-soil-src/soil/util.py", line 48, in get_download_context
    if download_data.task.failed():

  File "submodules/django-soil-src/soil/__init__.py", line 107, in task
    return Task.AsyncResult(self.task_id)

  File "/home/cchq/www/staging/python_env/local/lib/python2.7/site-packages/celery/app/task.py", line 751, in AsyncResult
    return self._get_app().AsyncResult(task_id, backend=self.backend,

  File "/home/cchq/www/staging/python_env/local/lib/python2.7/site-packages/celery/five.py", line 284, in __get__
    return self.__get.__get__(obj, type)()

  File "/home/cchq/www/staging/python_env/local/lib/python2.7/site-packages/celery/task/base.py", line 75, in backend
    return cls.app.backend

  File "/home/cchq/www/staging/python_env/local/lib/python2.7/site-packages/celery/local.py", line 143, in __getattr__
    return getattr(self._get_current_object(), name)

  File "/home/cchq/www/staging/python_env/local/lib/python2.7/site-packages/kombu/utils/__init__.py", line 322, in __get__
    value = obj.__dict__[self.__name__] = self.__get(obj)

  File "/home/cchq/www/staging/python_env/local/lib/python2.7/site-packages/celery/app/base.py", line 625, in backend
    return self._get_backend()

  File "/home/cchq/www/staging/python_env/local/lib/python2.7/site-packages/celery/app/base.py", line 444, in _get_backend
    return backend(app=self, url=url)

  File "/home/cchq/www/staging/python_env/local/lib/python2.7/site-packages/celery/backends/database/__init__.py", line 104, in __init__
    'Missing connection string! Do you have '

ImproperlyConfigured: Missing connection string! Do you have CELERY_RESULT_DBURI set to a real value?
```

**Comment by snopoke** [comment]
@gcapalbo any update on this?

**Comment by gcapalbo** [comment]
@snopoke I also can't seem to replicate that error locally, but according to http://celery.readthedocs.org/en/latest/django/first-steps-with-django.html, the update to localsettings you made is the right one to have.  (I tested locally with the old way and this new way, and they both seem to work and write to the db tables for some reason. I also verified that all the versions of the packages are right on staging.)

After making that change on staging, were you able to do testing of the exports and such?  I've already done testing of the sms/reminder-related uses of celery.

As long as we're done with testing, I'd say then for the next deploy that we coordinate merging this and updating localsettings on prod to use:

    CELERY_RESULT_BACKEND = 'djcelery.backends.database:DatabaseBackend'

@snopoke @czue does that sound good?

**Comment by snopoke** [comment]
I was able to run form exports without any issue on staging so I think this is good to go.

**Comment by czue** [comment]
sweet. @dannyroberts @TylerSheffels do you want to merge this prior to deploy and make sure that we get the settings updated at the same time?

**Comment by dannyroberts** [comment]
That sounds good. The steps @TylerSheffels updating localsettings will require editing the private variables file as described on https://confluence.dimagi.com/display/internal/Ansible. I think I'm not staying too late today and have a meeting at 3, so good if you grab me 30 mins before deploy to go over the steps if you need help with that.

**Comment by TylerSheffels** [comment]
@dannyroberts will do.

**Pull request merged**